src/Components/Common/MarkdownPreview.tsx

```diff
@@ -0,0 +1,88 @@
+import React, { useEffect, useState } from "react";
+import ReactMarkdown from "react-markdown";
+import rehypeRaw from "rehype-raw";
+import { UserModel } from "../Users/models";
+import useQuery from "../../Utils/request/useQuery";
+import routes from "../../Redux/api";
+
+interface UserCardProps {
+  user: UserModel;
+}
+
+const UserCard: React.FC<UserCardProps> = ({ user }) => (
+  <div className="flex w-64 items-center space-x-3 rounded-lg bg-white p-3 shadow-lg">
+    <div className="flex h-10 w-10 items-center justify-center rounded-full bg-primary text-lg font-semibold text-white">
+      {user.first_name[0]}
+    </div>
+    <div className="space-y-0">
+      <h3 className="text-sm font-semibold text-gray-800">
+        {user.first_name} {user.last_name}
+      </h3>
+      <p className="text-xs text-gray-500">@{user.username}</p>
+      <p className="text-xs text-gray-500">{user.user_type}</p>
+    </div>
+  </div>
+);
+
+const MarkdownPreview: React.FC<{ markdown: string }> = ({ markdown }) => {
+  const [userCache, setUserCache] = useState<Record<string, UserModel>>({});
+  const facilityId = "81092ced-8720-44cb-b4c5-3f0ad0540153";
+
+  const { data: facilityUsers } = useQuery(routes.getFacilityUsers, {
+    pathParams: { facility_id: facilityId },
+  });
+
+  useEffect(() => {
+    if (facilityUsers?.results) {
+      const newCache: Record<string, UserModel> = {};
+      facilityUsers.results.forEach((user) => {
+        if (user.username) {
+          newCache[user.username] = user as UserModel;
+        }
+      });
+      setUserCache(newCache);
+    }
+  }, [facilityUsers]);
```

Comment:
@UdaySagar-Git the user cache is for showing user information for mentioned users right?

If going this way, we could consider having the cache state at app level instead in the component level.

But it'd be better for the notes serializer itself to serialize the user details from backend, maybe as a separate attribute like how GitHub's APIs are. The note object would look something like this:

```
body: string
mentionedUsers: Array<UserMention>
```

Reply:
yes ! will change that to get details of mentioned users directly while fetching notes

Reply:
check @UdaySagar-Git

Reply:
done